runtime/compiler/aarch64/codegen/ARM64JNILinkage.cpp

```diff
@@ -111,7 +111,57 @@ TR::Register *J9::ARM64::JNILinkage::pushJNIReferenceArg(TR::Node *child)
 
 void J9::ARM64::JNILinkage::adjustReturnValue(TR::Node *callNode, bool wrapRefs, TR::Register *returnRegister)
    {
-   TR_UNIMPLEMENTED();
+   TR::SymbolReference      *callSymRef = callNode->getSymbolReference();
+   TR::ResolvedMethodSymbol *callSymbol = callNode->getSymbol()->castToResolvedMethodSymbol();
+   TR_ResolvedMethod *resolvedMethod = callSymbol->getResolvedMethod();
+
+   // jni methods may not return a full register in some cases so need to get the declared
+   // type so that we sign and zero extend the narrower integer return types properly
+   TR::LabelSymbol *tempLabel = generateLabelSymbol(cg());
+
+   switch (resolvedMethod->returnType())
+      {
+      case TR::Address:
+         if (wrapRefs)
+            {
+            // unwrap the return object is non-null
```

Comment:
Perhaps you mean "unwrap **when** the return object is non-null" ?

Reply:
Yes. I corrected the comment.